code/datums/gamemode/misc_gamemode_procs.dm

```diff
@@ -310,6 +310,44 @@
 			roles += player.mind.assigned_role
 	return roles
 
+/proc/equip_syndiborg(mob/living/silicon/robot/traitor_mob, var/uses = 20, var/datum/role/traitor/role)
+	if (!istype(traitor_mob))
+		return
+	. = 1
+
+	// find a radio! toolbox(es), backpack, belt, headset
+	var/loc = ""
+	var/list/contents = recursive_type_check(traitor_mob, /obj/item/device)
+	var/obj/item/R = locate(/obj/item/device/radio/borg) in contents //Hide the uplink in the borg radio module
+	if (!R)
+		to_chat(traitor_mob, "Unfortunately, the Syndicate wasn't able to get you a radio.")
+		. = 0
+	else
+		var/obj/item/device/uplink/hidden/T
+		if (istype(R, /obj/item/device/radio/borg))
+			// generate list of radio freqs
+			var/obj/item/device/radio/borg/target_radio = R
+			var/freq = 1441
```

Comment:
Make the original frequency setup a separate proc instead of copypasting it from equip_tator

Reply:
actually why not tweak the original equip_traitor and make this proc a child of that?

Reply:
Done